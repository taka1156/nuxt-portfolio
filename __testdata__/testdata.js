const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_ICONS_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const dummyUrl = 'http://placehold.jp';
const dummyCopyrightUrl = 'http://placehold.jp';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

const dummyOutsideLink = { routeTo: dummyUrl };
const dummyRouteStr = { routeTo: '/' };
const dummyRouteObj = {
  routeTo: {
    name: 'sample-id',
    params: { id: 1 }
  }
};

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = i => ({
  name: `ダミー${i}`,
  to: '/test1',
  img: dummyImgUrl
});

const dummyCardTemplate = i => ({
  title: `ダミー${i}`,
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい'
});

const dummyCardLinkTemplate = i => ({
  title: `ダミー${i}`,
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL
});

const dummyExamsTemplate = i => {
  return `ダミー${i}`;
};

const dummySnsIconTemplate = i => ({
  name: `ダミー${i}`,
  img: dummyImgUrl,
  link: dummyUrl
});

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーイメージ'
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, i => dummyNavTemplate(i));

const dummyCards = dummyFactory(MAX_CARD_DATA, i => dummyCardTemplate(i));

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummySnsIcons = dummyFactory(MAX_ICONS_DATA, i => dummySnsIconTemplate(i));

const dummyExams = {
  title: 'ダミー',
  values: dummyFactory(MAX_PROFILE_DATA, () => dummyExamsTemplate())
};

const dummyProfile = {
  img: dummyImgUrl,
  introduce:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。'
};

const dummyLogo = 'ダミー';

export {
  dummyOutsideLink,
  dummyRouteStr,
  dummyRouteObj,
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfile,
  dummyExams,
  dummySnsIcons,
  dummyCopyrightUrl
};
