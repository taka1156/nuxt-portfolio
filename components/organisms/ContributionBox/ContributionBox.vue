<template>
  <div class="contribution-box">
    <base-img :size="`free`" :img-url="imgUrl" :img-alt="imgAlt" />
  </div>
</template>

<script>
import BaseImg from '../../atoms/BaseImg/BaseImg';

export default {
  name: 'ContributionBox',
  components: {
    BaseImg
  },
  props: {
    /**
     * 画像のURL
     */
    imgUrl: {
      type: String,
      required: true
    },
    /**
     * 画像が表示できない時に使用する代替テキスト
     * (alt)
     */
    imgAlt: {
      type: String,
      required: true
    }
  }
};
</script>

<style scoped>
::v-deep .base-img--extend {
  width: 900px;
  object-fit: cover;
}

/* gh-chart */
.contribution-box {
  width: 85%;
  margin: 0 auto;
  overflow-x: scroll;
  border: 0.5px solid gray;
  border-radius: 5px;
}
</style>
