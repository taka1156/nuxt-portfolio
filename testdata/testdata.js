const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
