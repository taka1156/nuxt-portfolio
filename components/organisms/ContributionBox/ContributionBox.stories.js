import ContributionBox from './ContributionBox.vue';
import { dummyImg } from '@/__testdata__/testdata.js';

export default {
  title: 'Organisms/ContributionBox',
  component: ContributionBox
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { ContributionBox },
  template: '<contribution-box v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = dummyImg;
