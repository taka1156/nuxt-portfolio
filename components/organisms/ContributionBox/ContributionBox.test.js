import { shallowMount } from '@vue/test-utils';
import ContributionBox from './ContributionBox.vue';
import { dummyImg } from '@/__testdata__/testdata.js';

describe('contributionBox', () => {
  const contributionBox = propsData =>
    shallowMount(ContributionBox, {
      propsData: {
        ...propsData
      }
    });

  it('ContributionBox初期値: imgUrl, imgAlt', () => {
    const wrapper = contributionBox(dummyImg);
    // imgUrl
    expect(wrapper.vm.$options.props.imgUrl.required).toBe(true);
    expect(wrapper.vm.imgUrl).toBe(dummyImg.imgUrl);
    // imgAlt
    expect(wrapper.vm.$options.props.imgAlt.required).toBe(true);
    expect(wrapper.vm.imgAlt).toBe(dummyImg.imgAlt);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});
