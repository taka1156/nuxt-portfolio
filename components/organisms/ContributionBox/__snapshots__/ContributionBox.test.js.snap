// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contributionBox ContributionBox初期値: imgUrl, imgAlt 1`] = `
"<div class=\\"contribution-box\\">
  <base-img-stub imgurl=\\"http://placehold.jp/150x150.png\\" imgalt=\\"ダミーイメージ\\" size=\\"free\\"></base-img-stub>
</div>"
`;
